cluster/gce/gci/configure-helper_test.go

```diff
@@ -0,0 +1,215 @@
+	"strings"
+
+	"testing"
+
+	"os"
+	"path/filepath"
+
+	"encoding/base64"
+	"io/ioutil"
+
+	"k8s.io/api/core/v1"
+)
+
+const (
+	// Template for defining the environment state of configure-helper.sh
+	// The environment of configure-helper.sh is initially configured via kube-env file. However, as deploy-helper
+	// executes new variables are created. ManifestTestCase does not care where a variable came from. However, future
+	// test scenarios, may require such a distinction.
+
+	// The list of variables is, by no means, complete - this is what is required to run currently defined tests.
+	deployHelperEnv = `
+readonly KUBE_HOME={{.KubeHome}}
+readonly KUBE_API_SERVER_LOG_PATH=${KUBE_HOME}/kube-apiserver.log
+readonly KUBE_API_SERVER_AUDIT_LOG_PATH=${KUBE_HOME}/kube-apiserver-audit.log
+readonly CLOUD_CONFIG_OPT=--cloud-config=/etc/gce.conf
+readonly CA_CERT_BUNDLE_PATH=/foo/bar
+readonly APISERVER_SERVER_CERT_PATH=/foo/bar
+readonly APISERVER_SERVER_KEY_PATH=/foo/bar
+readonly APISERVER_CLIENT_CERT_PATH=/foo/bar
+readonly CLOUD_CONFIG_MOUNT="{\"name\": \"cloudconfigmount\",\"mountPath\": \"/etc/gce.conf\", \"readOnly\": true},"
+readonly CLOUD_CONFIG_VOLUME="{\"name\": \"cloudconfigmount\",\"hostPath\": {\"path\": \"/etc/gce.conf\", \"type\": \"FileOrCreate\"}},"
+readonly DOCKER_REGISTRY="k8s.gcr.io"
+readonly ENABLE_LEGACY_ABAC=false
+readonly ETC_MANIFESTS=${KUBE_HOME}/etc/kubernetes/manifests
+readonly KUBE_API_SERVER_DOCKER_TAG=v1.11.0-alpha.0.1808_3c7452dc11645d-dirty
+readonly LOG_OWNER_USER=$(whoami)
+readonly LOG_OWNER_GROUP=$(id -gn)
+ENCRYPTION_PROVIDER_CONFIG={{.EncryptionProviderConfig}}
+ENCRYPTION_PROVIDER_CONFIG_PATH={{.EncryptionProviderConfigPath}}
+readonly ETCD_KMS_KEY_ID={{.ETCDKMSKeyID}}
+`
+	kubeApiServerManifestFileName = "kube-apiserver.manifest"
+	kmsPluginManifestFileName     = "kms-plugin-container.manifest"
+	kubeApiServerStartFuncName    = "start-kube-apiserver"
+
+	// Position of containers within a pod manifest
+	kmsPluginContainerIndex        = 0
+	apiServerContainerIndexNoKMS   = 0
+	apiServerContainerIndexWithKMS = 1
+
+	//	command": [
+	//   "/bin/sh", - Index 0
+	//   "-c",      - Index 1
+	//   "exec /usr/local/bin/kube-apiserver " - Index 2
+	execArgsIndex = 2
+
+	socketVolumeMountIndexKMSPlugin = 1
+	socketVolumeMountIndexApiServer = 0
+)
+
+type kubeApiServerEnv struct {
+	KubeHome                     string
+	EncryptionProviderConfig     string
+	EncryptionProviderConfigPath string
+	ETCDKMSKeyID                 string
+}
+
+type kubeApiServerManifestTestCase struct {
+	*ManifestTestCase
+	apiServerContainer v1.Container
+	kmsPluginContainer v1.Container
+}
+
+func NewKubeApiServerManifestTestCase(t *testing.T) *kubeApiServerManifestTestCase {
+	c := &kubeApiServerManifestTestCase{}
+	c.ManifestTestCase = NewManifestTestCase(kubeApiServerManifestFileName, kubeApiServerStartFuncName, []string{kmsPluginManifestFileName}, t)
+
+	return c
+}
+
+func (c *kubeApiServerManifestTestCase) getContainers() {
```

Comment:
I'd expect this func to return something.
IMO, the whole kubeApiServerManifestTestCase type can be removed and its methods can be just regular helper funcs.
So this would be
func getContainers(tc *manifestTestCase) (v1.Container, v1.Container)